### ISSUE TITLE
[Feat]: Redesign Profile Page to Dashboard Layout

### ISSUE THREAD

**Issue by Titus210** [maniyaom/Movie-Ticket-Booking#123]
### Feature Description

Redesign Profile Page to Dashboard Layout

## Issue Overview
The current profile page design is cluttered with all information being displayed in a single view. To improve user experience, navigation, and access to different sections, this issue proposes a complete redesign of the Profile Page into a dashboard layout. The new layout will include separate tabs or sections for easy access to personal information, transactions, wallet details, tickets, and account settings.

### Dashboard Overview:
Implement a sidebar navigation menu that includes links to:
- Profile
- Tickets
- Wallet
- Transactions
- Logout
Display basic user information (profile picture, name, and location) in the sidebar.

## Profile Section:

- Separate sections for personal information and security details.
- Implement an "Edit" button to enable modification of profile information such as name, email, and phone number.
- Include a form for changing the account password under the security section.

## Transactions Section:

- Display all transactions in a table format, showing payment details, amount, and date.
- Implement a filter option to allow users to search or sort transactions.

## Wallet Section:

- Display the current wallet balance.
- Provide options to add funds or view wallet history.
## Tickets Section:

- Display booked tickets with event details and seat numbers.
- Implement a button for downloading or viewing ticket details.

### Set up Sidebar Navigation:

- [ ] Create a sidebar component with navigation links to all the relevant sections.
- [ ] Highlight the active section based on the user's navigation.

### Implement Profile Section:

- [ ] Separate the personal information and security details.
- [ ] Add input fields for profile information and a button to submit changes.
- [ ] Create a separate area for updating the password.

### Create Transactions Section:

- [ ] Set up a table component to display transaction history.
- [ ] Implement pagination and sorting features.
- [ ] Create a filter bar for searching transactions based on date, amount, or type.

###  Build Wallet Section:

- [ ] Display wallet balance and recent wallet-related activities.
- [ ] Create a form for adding funds.

### Create Tickets Section:

- [ ] Display user tickets in a card format with seat numbers and event details.
- [ ] Implement a "Download Ticket" button for each ticket.

@maniyaom I would like to work on this issue and redesign the profile page. Please assign it to me if possible.

## Initial Design view

![image](https://github.com/user-attachments/assets/cbeef9b5-bcb3-4f4e-a88f-68d629a21b4e)

## Proposed Design

1. Transactions Section

![Transactions (1)](https://github.com/user-attachments/assets/c3eb7d38-a91d-42c3-ae25-f4a6f4e5dbfc)

2. Tickets Section

![Tickets Page](https://github.com/user-attachments/assets/a3126c10-8858-4051-b720-16088e84f3fc)

3. Profile Section
![Profile page](https://github.com/user-attachments/assets/47e58684-7c40-44fc-975b-08df488cf5ac)

### By submitting this issue, you agree to

- [X] I agree to follow this project's Code of Conduct
- [X] I’ve read all the issues in the issue section, and this feature has not been proposed yet.

### Other Information

- [X] I'm a GSSoC '24 Extended contributor
- [X] I'm a Hacktoberfest '24 contributor
- [X] I am willing to work on this feature (optional)

**Comment by maniyaom** [comment]
@Titus210 have you already created this design and is it responsive?

**Comment by Titus210** [comment]
@maniyaom , Yes i have done the design, that is the desktop view design , currently working on mobile view design. It is responsive. and will be implementing it as a responsive design. 
[Figma design](https://www.figma.com/design/HaHHtTXekpfd6OFBS82dFH/Girlscript-Summer-of-code-%7C-Ticketify?node-id=0-1&t=9uI7qCwKaqQYflmD-1)

**Comment by maniyaom** [comment]
@Titus210 Okay looks good, will you able to implement it in tailwind CSS?

**Comment by Titus210** [comment]
@maniyaom  Yes...everything will be done in react tailwind css, incase i get stuck..will seek out for help. Thanks

**Comment by maniyaom** [comment]
@Titus210 any update to this?

**Comment by Titus210** [comment]
@maniyaom still working on the project. I unfortunately ran in some issues while trying to install the packages in the node modules that really took alot of my time *still did not manage to install this repo on my system. [example of issue](https://stackoverflow.com/questions/71282206/github-dependabot-alert-inefficient-regular-expression-complexity-in-nth-check) but I decided to create a new project to hold the initial implementation the code. I'm currently finalizing the dashbord implementation then reinsert the sections in our initial repo. Will update the repo by 4.00 am EAT.
